solana/write-account/src/lib.rs

```diff
@@ -124,6 +199,13 @@ fn handle_copy_null(dst: &mut [u8], src: &[u8]) -> Result {
         .ok_or(ProgramError::AccountDataTooSmall)
 }
 
+#[derive(BorshDeserialize, BorshSerialize, Debug)]
+pub struct State {
+    pub seed: Vec<u8>,
+    pub bump: u8,
+    pub data: Vec<u8>,
+}
```

Comment:
If we’re using borsh for this we might just as well use borsh for the entire instruction.